### PR TITLE
Branch a assertions

### PR DESCRIPTION
**Pull request by ong-hongkai**

### CODE REVIEW

File: src/main/java/winston/Event.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * A class extension of Task representing a deadline task.
+ */
+public class Event extends Task {
+    private final String type;
+    private final LocalDate dateAndTime;
+    
+    /**
+     * Constructor for Event
+     *
+     * @param description is the string of the description of the given task.
```
Comment (pcgiang) [inline src/main/java/winston/Event.java:15]: I think you missed out other parameters?

File: src/main/java/winston/InvalidCommand.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     */
+    public InvalidCommand(TaskList taskList) {
+        super(taskList);
+        this.errorMessage = "Invalid Command";
+    }
+
+    /**
+     * Constructor for InvalidCommand
+     *
+     * @param taskList the taskList with the list of tasks desired
```
Comment (pcgiang) [inline src/main/java/winston/InvalidCommand.java:22]: You may want to add in description for the other parameter too?

File: src/main/java/winston/Parser.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        this.taskList = taskList;
+    }
+
+    /**
+     * Method that parses the given user input.
+     * 
+     * @param str user input.
+     * @return A command depending on the user input.
+     */
+    public Command parse(String str) {
```
Comment (pcgiang) [inline src/main/java/winston/Parser.java:28]: I like that all the if and else if blocks are structured nicely, making very easy to trace the code! Good job!! 👍

File: src/main/java/winston/TaskList.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+     * @param position An integer corresponding to the task you wish to complete.
+     */
+    public void markTask(int position) {
+        list.get(position - 1).setComplete();
+        assert (storage.saveToFile(this));
+        storage.saveToFile(this);
+    }
+    
+    /**
+     * Removing a Task from the arraylist based on position.
```
Comment (pcgiang) [inline src/main/java/winston/TaskList.java:57]: You may want to change it to 'Remove a Task...' instead of 'Removing a Task...'

**Review by pcgiang — COMMENTED** [review]
LGTM! Overall, the code is very well structured and easy to understand. Naming and javadoc convention are adhered for the most part. Naming of variables and functions are also very clear and is not misleading in any way.

File: src/main/java/winston/AddDeadlineCommand.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package winston;
+
+/**
+ * Represents the deadline command from the parent abstract class Command.
+ */
+public class AddDeadlineCommand extends Command {
+    private final String description;
```
Comment (ykwei7) [inline src/main/java/winston/AddDeadlineCommand.java:7]: `private final String description;`
Not really related to coding standard violations, but there might be a need to edit the task naming hence I feel that setting it as final may not be ideal.

File: src/main/java/winston/DeadLine.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        return this.type + "," + super.saveFormat() + "," + this.dueDate;
+    }
+
+    /**
+     *  Method to convert information from object instance into a more readable format.
+     *   
+     * @return a string with the type, completion status, task description and due date.
+     */
+    @Override
+    public String toString() {
```
Comment (ykwei7) [inline src/main/java/winston/DeadLine.java:54]: Maybe you can try `String.format("[ %s ] %s (Due:  %s))", type, super.toString(), DateHandler.convertDate(dueDate)`

This improves readability for the string you intend to print in my opinion

File: src/main/java/winston/DeleteCommand.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    /**
+     * Deletes a task from the TaskList based on the index provided and prints number of uncompleted tasks remaining.
+     */
+    @Override
+    public String run() {
+        super.taskList.deleteTask(this.index);
+        String result =  Ui.printDontWorry() + Ui.printTasksLeft(super.taskList.numberOfIncompleteTasks());
+        assert(!result.equals(""));
```
Comment (ykwei7) [inline src/main/java/winston/DeleteCommand.java:21]: Good use of assertions to check for an empty task! 👍

**Review by ykwei7 — COMMENTED** [review]
Overall looks good!

**Comment by damithc** [comment]
Closing this as you already have another PR sent to this repo ...

**Pull request closed without merging**